packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/proxies/dataservice/CodeProxy.java

```diff
@@ -0,0 +1,56 @@
+package software.uncharted.terarium.hmiserver.proxies.dataservice;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+import software.uncharted.terarium.hmiserver.annotations.LogRestClientTime;
+import software.uncharted.terarium.hmiserver.models.dataservice.code.Code;
+import software.uncharted.terarium.hmiserver.models.dataservice.dataset.PresignedURL;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.Response;
+import java.util.List;
+
+/**
+ * Proxies the TDS code endpoints
+ */
+@RegisterRestClient(configKey = "data-service")
+@Path("/code")
+@Produces("application/json")
+@Consumes("application/json")
+public interface CodeProxy {
+
+    	@GET
+		@LogRestClientTime
+    	List<Code> getCodes( @DefaultValue("100") @QueryParam("page_size") Integer pageSize, @DefaultValue("0") @QueryParam("page") Integer page);
+
+		@POST
+		@LogRestClientTime
+		Code createCode(JsonNode code);
+
+		@GET
+		@Path("/{id}")
+		@LogRestClientTime
+		Code getCode(@PathParam("id") String codeId);
+
+		@PUT
```

Comment:
We should be consistent, sometimes it's PATCH, sometimes PUT.

Reply:
We are using `PUTs` for every endpoint set.  The `PATCH` endpoint caused problems so we only implemented it on the datasets.